storage/Devices/BcacheCsetImpl.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2016 SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#include <boost/regex.hpp>
+
+#include "storage/Utils/XmlFile.h"
+#include "storage/Utils/StorageTmpl.h"
+#include "storage/Utils/StorageDefines.h"
+#include "storage/SystemInfo/SystemInfo.h"
+#include "storage/Devices/BcacheCsetImpl.h"
+#include "storage/Devices/BlkDeviceImpl.h"
+#include "storage/Holders/User.h"
+
+
+namespace storage
+{
+
+    using namespace std;
+
+
+    const char* DeviceTraits<BcacheCset>::classname = "BcacheCset";
+
+
+    BcacheCset::Impl::Impl(const xmlNode* node)
+	: Device::Impl(node), uuid()
+    {
+	getChildValue(node, "uuid", uuid);
+    }
+
+
+    void
+    BcacheCset::Impl::save(xmlNode* node) const
+    {
+	Device::Impl::save(node);
+
+	setChildValue(node, "uuid", uuid);
+    }
+
+
+    bool
+    BcacheCset::Impl::is_valid_uuid(const string& uuid)
+    {
+	static boost::regex uuid_regex("[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}",
```

Comment:
I tought about that as well but at least LVM uses different UUIDs, e.g. "OMPzXF-m3am-1zIl-AVdQ-i5Wx-tmyN-cevmRn".